versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoDbStore.java

```diff
@@ -0,0 +1,215 @@
+    MongoCollection<InternalCommitMetadata> commitMetadataMongoCollection =
+        mongoDatabase.getCollection(commitMetadataCollection, InternalCommitMetadata.class);
+  }
+
+  /**
+   * Closes the connection this manager creates to a database. If the connection is already closed this method has
+   * no effect.
+   */
+  @Override
+  public void close() {
+  }
+
+  @Override
+  public void load(LoadStep loadstep) throws ReferenceNotFoundException {
+    throw new UnsupportedOperationException();
+  }
+
+  private List<ListMultimap<String, LoadOp<?>>> paginateLoads(LoadStep loadStep, int size) {
+    List<ListMultimap<String, LoadOp<?>>> paginated = new ArrayList<>();
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public <V> boolean putIfAbsent(ValueType type, V value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <V> void put(ValueType type, V value, Optional<ConditionExpression> conditionUnAliased) {
+    Preconditions.checkArgument(type.getObjectClass().isAssignableFrom(value.getClass()),
+        "ValueType %s doesn't extend expected type %s.", value.getClass().getName(), type.getObjectClass().getName());
+    final Map<String, Entity> attributes = type.getSchema().itemToMap(value, true);
+    LOGGER.info("Mongo attributes: " + attributes.toString());
+    // TODO ensure that calls to mongoDatabase.createCollection etc are surrounded with try-catch to detect
+    //  com.mongodb.MongoSocketOpenException
+    l2MongoCollection = mongoDatabase.getCollection(l2Collection, L2.class);
+    LOGGER.info("Connected to ", mongoDatabase.toString());
+    LOGGER.info("ValueType: " + type.toString() + " Value: " + ((L2)value).toString());
+    if (type.equals(ValueType.L2)) {
+      LOGGER.info("About to insert value");
+      l2MongoCollection.insertOne((L2)value);
+      LOGGER.info("Inserted value");
+    }
+  }
+
+  @Override
+  public boolean delete(ValueType type, Id id, Optional<ConditionExpression> condition) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void save(List<SaveOp<?>> ops) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <V> V loadSingle(ValueType valueType, Id id) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <V> Optional<V> update(ValueType type, Id id, UpdateExpression update, Optional<ConditionExpression> condition)
+      throws ReferenceNotFoundException {
+    throw new UnsupportedOperationException();
+  }
+
+  private final boolean tableExists(String name) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public Stream<InternalRef> getRefs() {
+    throw new UnsupportedOperationException();
+  }
+
+  @VisibleForTesting
+  public MongoDatabase getMongoDatabase() {
```

Comment:
Even if this is exposed for testing, (not private), you should use package private)

Reply:
public accessor has been removed.